images/virtualization-artifact/pkg/controller/vd/internal/life_cycle.go

```diff
@@ -103,10 +123,13 @@ func (h LifeCycleHandler) Handle(ctx context.Context, vd *virtv2.VirtualDisk) (r
 		}
 	}
 
-	requeue, err := ds.Sync(ctx, vd)
-	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to sync virtual disk data source %s: %w", ds.Name(), err)
+	if storageClassReadyCondition.Status == metav1.ConditionTrue || readyCondition.Status == metav1.ConditionTrue {
```

Comment:
The condition `readyCondition.Status == metav1.ConditionTrue` seems redundant; is it really necessary here? If so, it would be better to add a comment explaining why.